src/context/cart/cart-context.js

```
@@ -0,0 +1,119 @@
+        cartTotalItems: newTotalItems,
+        cartTotalPrice: newTotalPrice,
+      };
+    }
+
+    case 'REMOVE_FROM_CART': {
+      const productToRemove = action.payload.product;
+
+      let newCartProducts = state.cartProducts.filter(
+        (product) => product.id !== productToRemove.id
+      );
+
+      console.log(newCartProducts);
+
+      let newTotalPrice =
+        Number(state.cartTotalPrice) -
+        Number(productToRemove.quantity) * Number(productToRemove.price);
+
+      const newTotalItems = state.cartTotalItems - 1;
+
+      return {
+        ...state,
+        cartProducts: newCartProducts,
+        cartTotalItems: newTotalItems,
+        cartTotalPrice: newTotalPrice,
+      };
+    }
+
+    case 'INC_ITEM_QUANTITY': {
+      const productId = action.payload.id;
+
+      const index = state.cartProducts.findIndex(
+        (product) => product.id === productId
+      );
+
+      let newProduct = { ...state.cartProducts[index] };
+      newProduct.quantity += 1;
+
+      let newTotalPrice =
+        Number(state.cartTotalPrice) + Number(newProduct.price);
+
+      let newCartProducts = [...state.cartProducts];
+      newCartProducts[index] = newProduct;
+
+      return {
+        ...state,
+        cartProducts: newCartProducts,
+        cartTotalPrice: newTotalPrice,
+      };
+    }
+    case 'DEC_ITEM_QUANTITY': {
+      const productId = action.payload.id;
+
+      const index = state.cartProducts.findIndex(
+        (product) => product.id === productId
+      );
+
+      let newProduct = { ...state.cartProducts[index] };
+      newProduct.quantity -= 1;
+
+      let newTotalPrice =
+        Number(state.cartTotalPrice) - Number(newProduct.price);
+
+      let newCartProducts = [...state.cartProducts];
+      newCartProducts[index] = newProduct;
+
+      return {
+        ...state,
+        cartProducts: newCartProducts,
+        cartTotalPrice: newTotalPrice,
+      };
+    }
+
+    default:
+      break;
```

You can throw a new error or just return the state as it is as a default. This helps if there is a typo while writing a new dispatch